src/test/java/org/assertj/core/internal/throwables/Throwables_assertHasRootCause_Test.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2018 the original author or authors.
+ */
+package org.assertj.core.internal.throwables;
+
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.error.ShouldHaveCause.shouldHaveCause;
+import static org.assertj.core.error.ShouldHaveNoCause.shouldHaveNoCause;
+import static org.assertj.core.error.ShouldHaveRootCause.shouldHaveRootCause;
+import static org.assertj.core.test.TestData.someInfo;
+import static org.assertj.core.test.TestFailures.failBecauseExpectedAssertionErrorWasNotThrown;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.mockito.Mockito.verify;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.ThrowablesBaseTest;
+import org.junit.jupiter.api.Test;
+
+public class Throwables_assertHasRootCause_Test extends ThrowablesBaseTest {
+
+  private static final String EXCEPTION_MESSAGE = "invalid arg";
+
+  private Throwable rootCause = new IllegalArgumentException(EXCEPTION_MESSAGE);
+  private Throwable throwableWithRootCause = new Throwable(new RuntimeException(rootCause));
+
+  @Test
+  public void should_pass_if_cause_has_expected_type_and_message() {
+    throwables.assertHasRootCause(someInfo(), throwableWithRootCause, new IllegalArgumentException(EXCEPTION_MESSAGE));
+  }
+
+  @Test
+  public void should_pass_if_actual_has_no_cause_and_expected_cause_is_null() {
+    throwables.assertHasRootCause(someInfo(), new Throwable(), null);
+  }
+
+  @Test
+  public void should_fail_if_actual_is_null() {
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> throwables.assertHasRootCause(someInfo(), null, new Throwable()))
+      .withMessage(actualIsNull());
+  }
+
```

Comment:
it would be great if all the `should_fail_...` tests could follow an approach like this:
```
  @Test
  void should_fail_if_actual_has_a_message_containing_the_given_content() {
    // GIVEN
    String content = "message";
    // WHEN
    expectAssertionError(() -> throwables.assertHasMessageNotContaining(INFO, actual, content));
    // THEN
    verify(failures).failure(INFO, shouldNotContain(actual.getMessage(), content), actual.getMessage(), content);
  }
```
the code will be lighter and more readable.

The code you have copied dates from a time such java 8 based capabilities were not available.